### PR TITLE
Dumbness shouldn't drive...

### PR DESCRIPTION
**Pull request by MystLeissa**
Prevents people with the Dumb Trait from piloting mecha

## About The Pull Request

Split off from RC-1 this isolation contains the idea that it's a slight oversight that people with the dumb trait can drive a highly complicated machine. We'll fix that
 (Note: I have no IDEA where to get the TRAIT_DUMB from naturally but I used it in my lockable schoolgirl branch to interact with this as well as a few other things that the flag came up on)

## Why It's Good For The Game

Piloting a Mech Suit shouldn't be so simple a dumb person can do it.

## Changelog
:cl:
tweak: Dumb People Can't Drive Mecha anymore.
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
Would be cooler if having the dumb trait automatically applied the uncontrollable movement that you normally get from mech malfunctions.

**Comment by MystLeissa** [comment]
> Would be cooler if having the dumb trait automatically applied the uncontrollable movement that you normally get from mech malfunctions.

Was thinking about that myself - but I'm not sure that's within my skill level...it isn't exactly as easy as setting an my_pos + <frand(3),frand(3),frand(3)>

**Comment by HomocidlePancakes** [comment]
Should do the same to savages

**Review by Fikou — CHANGES_REQUESTED** [review]
this has bad spelling and random whitespaces for no reason please fix

**Comment by nemvar** [comment]
@HomocidlePancakes Savages trait isn't real.

**Comment by MystLeissa** [comment]
> this has bad spelling and random whitespaces for no reason please fix

where are words misspelled? As for white space - that's a matter of preference - I used tab - and any other whitespace is from me hitting enter so it's more of a linebreak.

File: code/game/mecha/mecha.dm [hunk 1]
```diff
@@ -949,8 +954,7 @@
 			return mmi_moved_inside(mmi_as_oc, user)
 		else
 			to_chat(user, "<span class='warning'>Occupant detected!</span>")
-	else
```
Comment (MrDoomBringer) [inline code/game/mecha/mecha.dm]: Why did this else get deleted?

**Comment by kingofkosmos** [comment]
Could this be applied through relaymove to driving anything at all?
Also text could have a little fluff like "You have no idea how to operate \the [src]!"

File: code/game/mecha/mecha.dm (hunk 1 above)
Comment (MystLeissa) [reply code/game/mecha/mecha.dm]: it was not needed, I'd have to reference the entire proc to know for sure - but I'd say the reasoning was the "Protip: Early Returns are preferred" reference in the wiki.

Comment (MystLeissa) [reply code/game/mecha/mecha.dm]: Actually your right - I fixed it  in the latest commit; I'm pretty sleepless..

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -892,9 +892,11 @@
 		to_chat(user, "<span class='warning'>You can't enter the exosuit with other creatures attached to you!</span>")
 		log_message("Permission denied (Attached mobs).", LOG_MECHA)
 		return
-
+	if(HAS_TRAIT(user,TRAIT_DUMB))
+		to_chat(user, "<span class='warning'>This Mech will not let dumb people drive it</span>")
```
Comment (Akrilla) [inline code/game/mecha/mecha.dm]: Mech isn't capitalized, also needs some sort of punctuation at the end.

File: code/game/mecha/mecha.dm [hunk 3]
```diff
@@ -904,11 +906,12 @@
 			to_chat(user, "<span class='warning'>You can't enter the exosuit while buckled.</span>")
 		else if(user.has_buckled_mobs())
 			to_chat(user, "<span class='warning'>You can't enter the exosuit with other creatures attached to you!</span>")
+		else if(HAS_TRAIT(user,TRAIT_DUMB))
+			to_chat(user,"<span class='warning'>You can't drive dummy</span>")
```
Comment (Akrilla) [inline code/game/mecha/mecha.dm]: same here

File: code/game/mecha/mecha.dm [hunk 4]
```diff
@@ -904,11 +906,12 @@
...
 			to_chat(user, "<span class='warning'>You can't enter the exosuit with other creatures attached to you!</span>")
+		else if(HAS_TRAIT(user,TRAIT_DUMB))
+			to_chat(user,"<span class='warning'>You can't drive dummy</span>")
 		else
 			moved_inside(user)
-	else
-		to_chat(user, "<span class='warning'>You stop entering the exosuit!</span>")
-	return
+			return
+		to_chat(user, "<span class='warning'>You stop entering the exosuit!</span>");
```
Comment (Akrilla) [inline code/game/mecha/mecha.dm]: You've removed a return here, so this line will _always_ be said, which I don't imagine is intended.

Comment (MrPerson) [reply code/game/mecha/mecha.dm]: The return is if the move into the mecha was successful. By this point, the only case left is failure of some type. It's inverted logic, but lolmechs

**Comment by ATH1909** [comment]
Don't we already have a "dumb" brain trauma? Why not just check for that (or the clumsy trait/mutation)?

**Comment by Detective-Google** [comment]
> Don't we already have a "dumb" brain trauma? Why not just check for that (or the clumsy trait/mutation)?

if you check for clumsy clowns couldn't pilot H.O.N.K.

**Pull request closed without merging**